jerry-debugger/jerry_client.py

```diff
@@ -0,0 +1,327 @@
+                _scroll_direction(self.debugger, "up")
+            elif key == 's\n':
+                _scroll_direction(self.debugger, "down")
+            elif key == 'q\n':
+                break
+            else:
+                print("Invalid key")
+
+    def do_continue(self, _):
+        """ Continue execution """
+        self.debugger.get_continue()
+        self.stop = True
+        if not self.debugger.non_interactive:
+            print("Press enter to stop JavaScript execution.")
+    do_c = do_continue
+
+    def do_finish(self, _):
+        """ Continue running until the current function returns """
+        self.debugger.finish()
+        self.stop = True
+    do_f = do_finish
+
+    def do_dump(self, args):
+        """ Dump all of the debugger data """
+        if args:
+            print("Error: No argument expected")
+        else:
+            pprint(self.debugger.function_list)
+
+    def do_eval(self, args):
+        """ Evaluate JavaScript source code """
+        self.debugger.eval(args)
+        self.stop = True
+    do_e = do_eval
+
+    def do_memstats(self, _):
+        """ Memory statistics """
+        self.debugger.memstats()
+        self.stop = True
+    do_ms = do_memstats
+
+    def do_abort(self, args):
+        """ Throw an exception """
+        self.debugger.abort(args)
+        self.stop = True
+
+    def do_restart(self, _):
+        """ Restart the engine's debug session """
+        self.debugger.restart()
+        self.stop = True
+    do_res = do_restart
+
+    def do_throw(self, args):
+        """ Throw an exception """
+        self.debugger.throw(args)
+        self.stop = True
+
+    def do_exception(self, args):
+        """ Config the exception handler module """
+        print(self.debugger.exception(args))
+
+def _scroll_direction(debugger, direction):
+    """ Helper function for do_scroll """
+    debugger.src_offset_diff = int(max(math.floor(debugger.display / 3), 1))
+    if direction == "up":
+        debugger.src_offset -= debugger.src_offset_diff
+    else:
+        debugger.src_offset += debugger.src_offset_diff
+    print(debugger.print_source(debugger.display, debugger.src_offset))
+
+def src_check_args(args):
+    try:
+        line_num = int(args)
+        if line_num < 0:
+            print("Error: Non-negative integer number expected")
+            return -1
+
+        return line_num
+    except ValueError as val_errno:
+        print("Error: Non-negative integer number expected: %s" % (val_errno))
+        return -1
+
+# pylint: disable=too-many-branches,too-many-locals,too-many-statements
+def main():
+    args = jerry_client_ws.arguments_parse()
+
+    debugger = jerry_client_ws.JerryDebugger(args.address)
+
+    non_interactive = args.non_interactive
+
+    logging.debug("Connected to JerryScript on %d port", debugger.port)
+
+    prompt = DebuggerPrompt(debugger)
+    prompt.prompt = "(jerry-debugger) "
+    prompt.debugger.non_interactive = non_interactive
```

Comment:
This `non_interactive` variable is unnecessary. You can use `arg.non_interactive` directly.